src/vma/dev/qp_mgr.cpp

```diff
@@ -500,34 +512,40 @@ int qp_mgr::post_recv(mem_buf_desc_t* p_mem_buf_desc)
 	return 0;
 }
 
+static inline void mlx5_bf_copy(unsigned long long *dst, unsigned long long *src)
+{
+	COPY_64B_NT(dst, src);
+}
+
 void qp_mgr::mlx5_send(vma_ibv_send_wr *p_send_wqe)
 {
 	uintptr_t addr = 0;
 	uint32_t length = 0;
+	uint32_t lkey = 0;
 
 	addr = p_send_wqe->sg_list[0].addr;
+	length = p_send_wqe->sg_list[0].length;
+	lkey = p_send_wqe->sg_list[0].lkey;
 
-	/* Copy the first 16 bytes into the inline header */
-	memcpy((void *)(uintptr_t)m_sq_hot_wqe->eseg.inline_hdr_start,
-	       (void *)(uintptr_t)addr,
+	/* Copy the first bytes into the inline header */
+	memcpy((void *)m_sq_hot_wqe->eseg.inline_hdr_start,
+	       (void *)addr,
 	       MLX5_ETH_INLINE_HEADER_SIZE);
 
 	addr += MLX5_ETH_INLINE_HEADER_SIZE;
-	length = p_send_wqe->sg_list[0].length;
 	length -= MLX5_ETH_INLINE_HEADER_SIZE;
 
 	m_sq_hot_wqe->dseg.byte_count = htonl(length);
-	m_sq_hot_wqe->dseg.lkey = htonl(p_send_wqe->sg_list[0].lkey);
+	m_sq_hot_wqe->dseg.lkey = htonl(lkey);
 	m_sq_hot_wqe->dseg.addr = htonll(addr);
 
-	m_sq_wqe_idx_to_wrid[m_sq_hot_wqe_index] = (uintptr_t)p_send_wqe->wr_id;
-
-	++m_sq_wqe_counter;
-	volatile uintptr_t *dst = (volatile uintptr_t *)
-			((uintptr_t)m_sq_bf_reg + m_sq_bf_offset);
-
+	/*
+	 * Make sure that descriptors are written before
+	 * updating doorbell record and ringing the doorbell
+	 */
 	wmb();
 	*m_sq_db = htonl(m_sq_wqe_counter);
```

Comment:
In the original code m_sq_wqe_counter is already incremented while in your PR it is not. 
Please explain if 
```
*m_sq_db = htonl(m_sq_wqe_counter);
```
has the same value before/after your PR. Please justify this logic